pkg/generate/app/cmd/newapp.go

```diff
@@ -279,36 +292,68 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 		common := app.PipelineGroup{}
 		for _, ref := range group {
 
-			var pipeline *app.Pipeline
-			if ref.Input().ExpectToBuild {
-				glog.V(2).Infof("will use %q as the base image for a source build of %q", ref, ref.Input().Uses)
-				input, err := app.InputImageFromMatch(ref.Input().Match)
-				if err != nil {
-					return nil, fmt.Errorf("can't build %q: %v", ref.Input(), err)
-				}
-				strategy, source, err := app.StrategyAndSourceForRepository(ref.Input().Uses)
-				if err != nil {
-					return nil, fmt.Errorf("can't build %q: %v", ref.Input(), err)
-				}
-				if pipeline, err = app.NewBuildPipeline(ref.Input().String(), input, strategy, source); err != nil {
-					return nil, fmt.Errorf("can't build %q: %v", ref.Input(), err)
+			// TODO template and other individual matches are mutually exclusive,
+			// need to investigate the side-effects of having them together
+			if ref.Input().Match.Template != nil {
+				glog.V(2).Infof("will include objects from template %q", ref)
```

Comment:
@smarterclayton I wasn't sure about the correct strategy here:

1. Break down the template and handle each of its objects as one individual pipeline here in `buildPipelines`. I think this is the right approach, but I will need to figure out how to handle Services, Routes, etc.
2. Break down the template before, maybe in `validate` and make each separate object a `ComponentInput` (I think it's wrong).
3. Handle the Template as one single pipeline element returned by `buildPipelines` and break it down later.

Thoughts?